monitoring/uss_qualifier/scd/simulator/operational_intents_flight_authorisation_data_generator.py

```diff
@@ -0,0 +1,373 @@
+    
+        should_intersect = False 
+        if injection_number != 0:
+            coin_flip = random.choice([0,0,1]) # Geometry may or may not intersect, the first one will never intersect
+            should_intersect = coin_flip
+
+        geometry_generation_rule = GeometryGenerationRule(intersect_space = should_intersect)
+
+        flight_geometry = my_operational_intent_generator.generate_nominal_test_geometry(geometry_generation_rule= geometry_generation_rule, injection_number = injection_number)       
+        reference_time = my_operational_intent_generator.now.isoformat()
+    
+        flight_volume = my_operational_intent_generator.generate_astm_4d_volumes(raw_geometry = flight_geometry, altitude_of_ground_level_wgs_84 = altitude_of_ground_level_wgs_84)
+        
+        operational_intent_test_injection = generate_injection_operational_intent(astm_4d_volume = flight_volume)
+    
+        inject_flight_request = InjectFlightRequest(operational_intent= operational_intent_test_injection, flight_authorisation= flight_authorisation_data)
+        
+        expected_flight_authorisation_processing_result = 'Rejected' if generate_incorrect_data else 'Planned'
+        expected_operational_intent_processing_result= 'ConflictWithFlight' if should_intersect else 'Planned'
+
+        my_known_issues_acceptable_result_generator = KnownIssuesAcceptableResultFieldGenerator(expected_flight_authorisation_processing_result = expected_flight_authorisation_processing_result,expected_operational_intent_processing_result = expected_operational_intent_processing_result)
+        all_incorrect_result_details = my_known_issues_acceptable_result_generator.generate_known_issue_fields(incorrect_field= incorrect_field)
+
+        acceptable_results = my_known_issues_acceptable_result_generator.acceptable_results
+        
+        injection_target = InjectionTarget(uss_role = "Submitting USS")
+        
+        known_responses = KnownResponses(acceptable_results=acceptable_results, incorrect_result_details= all_incorrect_result_details)
+        
+        flight_injection_attempt = FlightInjectionAttempt(reference_time = reference_time, test_injection = inject_flight_request, known_responses = [known_responses],injection_target = injection_target)
+        
+        flight_injection_attempts.append(flight_injection_attempt)
+    
+    return flight_injection_attempts
+    
+
+def write_automated_test_to_disk(output_path:os.path, flight_injection_attempts: List[FlightInjectionAttempt], country_code="che") -> None:
+    """A function to write Flight injection attempts to disk so that they can be examined / used by other software like the test executor """
+        
+    output_directory = Path(output_path, country_code) 
```

Comment:
May I suggest to use locale instead of country_code in order to harmonize with the test executor and other components?

Reply:
Thanks! OK I have now switched to `locale` instead of `country code`, in the rid simulator, I had used country code, will make a note to switch it there as well